### PR TITLE
Service accounts: split user and service account database

### PR DESCRIPTION
**Pull request by Jguer**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/2849

**Special notes for your reviewer**:

### CODE REVIEW

**Review by gamab — APPROVED** [review]
LGTM, but I'd suggest asking the abstraction squad if that's ok with them as well :slightly_smiling_face:

**Review by kalleep — APPROVED** [review]
LGTM

**Review by IevaVasiljeva — APPROVED** [review]
All looks good to me! I also tried it out and didn't run into any issues.

The only thing that I'm wondering about is whether it's fine to remove the listing endpoint. But I guess SAs are in beta, so it's fine to make changes like that.

File: pkg/services/serviceaccounts/api/api.go [hunk 1]
```diff
@@ -55,7 +55,6 @@ func (api *ServiceAccountsAPI) RegisterAPIEndpoints(
 
 	auth := acmiddleware.Middleware(api.accesscontrol)
 	api.RouterRegister.Group("/api/serviceaccounts", func(serviceAccountsRoute routing.RouteRegister) {
-		serviceAccountsRoute.Get("/", auth(middleware.ReqOrgAdmin, accesscontrol.EvalPermission(serviceaccounts.ActionRead, serviceaccounts.ScopeAll)), routing.Wrap(api.ListServiceAccounts))
```
[the remark this answers is not in the document]
Comment (Jguer) [reply pkg/services/serviceaccounts/api/api.go:58]: Yes, this endpoint is unused, undocumented and duplicates a lot of logic in `/search`. We found it preferable to remove it and if needed we'll re-implement it sharing some consistency with search

File: pkg/services/serviceaccounts/api/api.go [hunk 2]
```diff
@@ -229,45 +207,26 @@ func (api *ServiceAccountsAPI) SearchOrgServiceAccountsWithPaging(c *models.ReqC
...
-	query := &models.SearchOrgUsersQuery{
-		OrgID:            c.OrgId,
-		Query:            c.Query("query"),
-		Page:             page,
-		Limit:            perPage,
-		User:             c.SignedInUser,
-		IsServiceAccount: true,
-	}
-	serviceAccounts, err := api.store.SearchOrgServiceAccounts(ctx, query)
+	serviceAccountSearch, err := api.store.SearchOrgServiceAccounts(ctx, c.OrgId, c.Query("query"), page, perPage, c.SignedInUser)
```
Comment (eleijonmarck) [inline pkg/services/serviceaccounts/api/api.go:210]: i really like this approach better!

Comment (eleijonmarck) [reply pkg/services/serviceaccounts/api/api.go:210]: this is the way going forward right?

Comment (Jguer) [reply pkg/services/serviceaccounts/api/api.go:210]: @eleijonmarck from what I've seen yes, it avoids hiding mandatory args in structures

**Pull request merged**